### PR TITLE
Additional tests for HintRetrieveRequest

### PR DESCRIPTION
**Pull request by iainmcgin**

### CODE REVIEW

File: protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java [hunk 1]
```diff
@@ -36,6 +38,77 @@
 public class HintRetrieveRequestTest {
 
     @Test
+    public void testFromAuthMethods() {
+        HintRetrieveRequest request = HintRetrieveRequest.fromAuthMethods(
+                AuthenticationMethods.EMAIL,
+                AuthenticationMethods.FACEBOOK);
+
+        assertThat(request.getAuthenticationMethods().size()).isEqualTo(2);
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.EMAIL);
```
Comment (dxslly) [inline protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java]: FYI, I think there is a containsExactly() method for set equality assertions.

File: protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java [hunk 2]
```diff
@@ -36,6 +38,77 @@
...
+        assertThat(request.getAuthenticationMethods()).contains(AuthenticationMethods.FACEBOOK);
+    }
+
+    @Test(expected=IllegalArgumentException.class)
+    public void testFromAuthMethods_emptySet() {
+        HintRetrieveRequest request = HintRetrieveRequest.fromAuthMethods(
+                Collections.emptySet());
+
```
Comment (dxslly) [inline protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java]: Test should have failed before these lines are run. Remove please.

File: protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java [hunk 3]
```diff
@@ -36,6 +38,77 @@
...
+    }
+
+    @Test
+    public void testBuilder_addTokenProvider_noInfo() {
+        String tokenProvider = "https://my.token.provider";
+        HintRetrieveRequest request = new HintRetrieveRequest.Builder(AuthenticationMethods.EMAIL)
+                .addTokenProvider(tokenProvider)
+                .build();
+
+        assertThat(request.getTokenProviders().containsKey(tokenProvider));
```
Comment (dxslly) [inline protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java]: Missing an assertion here.

File: protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java (hunk 1 above)
Comment (iainmcgin) [reply protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java]: Oooh, me gusta. Done.

File: protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java (hunk 2 above)
Comment (iainmcgin) [reply protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java]: I do love to copy paste. Done.

File: protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java (hunk 3 above)
Comment (iainmcgin) [reply protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java]: misplaced bracket, containsKey is an both a method on maps and on the map assertions :/

**Comment by iainmcgin** [comment]
PTAL

File: protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java [hunk 4]
```diff
@@ -17,6 +17,8 @@
 import static org.assertj.core.api.Java6Assertions.assertThat;
 
 import android.os.Parcel;
+import com.google.common.collect.ImmutableSet;
+import edu.emory.mathcs.backport.java.util.Collections;
 import java.util.HashMap;
```
Comment (dxslly) [inline protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java]: java/util/Collections?

Comment (iainmcgin) [reply protocol/javatests/org/openyolo/protocol/HintRetrieveRequestTest.java]: Weird. Thanks, Obama / Android Studio.

**Comment by iainmcgin** [comment]
@nerdyverde I'll wait for your review to follow the process.

**Pull request merged**